build.gradle.kts

```diff
@@ -83,35 +83,36 @@ fun generateTask(taskName: String, incubating: Boolean) {
     standardOutput = System.out
     executable = "docker"
 
-    var filter = if (incubating) "any" else "is_stable"
-    var classPrefix = if (incubating) "Incubating" else ""
+    var target = if (incubating) "incubating_java" else "java"
     val outputDir = if (incubating) "semconv-incubating/src/main/java/io/opentelemetry/semconv/incubating/" else "semconv/src/main/java/io/opentelemetry/semconv/"
-    val packageNameArg = if (incubating) "io.opentelemetry.semconv.incubating" else "io.opentelemetry.semconv"
-    val stablePackageNameArg = if (incubating) "io.opentelemetry.semconv" else ""
 
-    setArgs(listOf(
-        "run",
+    val file_args = if (org.gradle.internal.os.OperatingSystem.current().isWindows())
+      // Don't need to worry about file system permissions in docker.
+      listOf()
+    else {
+      // Make sure we run as local file user
+      val unix = com.sun.security.auth.module.UnixSystem() 
+      val uid = unix.getUid() // $(id -u $USERNAME)
+      val gid = unix.getGid() // $(id -g $USERNAME)
+      listOf("-u", "$uid:$gid")
+    }
+    val weaver_args = listOf(
         "--rm",
-        "-v", "$buildDir/semantic-conventions-${semanticConventionsVersion}/model:/source",
-        "-v", "$projectDir/buildscripts/templates:/templates",
-        "-v", "$projectDir/$outputDir:/output",
-        "otel/semconvgen:$generatorVersion",
-        "--yaml-root", "/source",
-        "--continue-on-validation-errors",
-        "code",
-        "--template", "/templates/SemanticAttributes.java.j2",
-        "--output", "/output/{{pascal_prefix}}${classPrefix}Attributes.java",
-        "--file-per-group", "root_namespace",
-        // Space delimited list of root namespaces to excluded (i.e. "foo bar")
-        "-Dexcluded_namespaces=ios aspnetcore signalr",
-        "-Dexcluded_attributes=messaging.client_id",
-        "-Dfilter=${filter}",
-        "-DclassPrefix=${classPrefix}",
-        "-Dpkg=$packageNameArg",
-        "-DstablePkg=$stablePackageNameArg"))
+        "--mount", "type=bind,source=$buildDir/semantic-conventions-${semanticConventionsVersion}/model,target=/home/weaver/source,readonly",
+        "--mount", "type=bind,source=$projectDir/buildscripts/templates,target=/home/weaver/templates,readonly",
+        "--mount", "type=bind,source=$projectDir/$outputDir,target=/home/weaver/target",
+        "otel/weaver:$generatorVersion",
+        "registry", "generate",
+        "--registry=/home/weaver/source",
+        "--templates=/home/weaver/templates",
+        "$target",
+        "/home/weaver/target/")
+
+    setArgs(listOf("run") + file_args + weaver_args)
   }
 }
 
+// TODO - With weaver we can generate both of this in one go, but let's refactor smaller pieces at a time.
```

Comment:
Meh.. we already have a `generateSemanticConventions` task which merged both into one. Also, it can be nice to have the ability to isolate to only generate stable or incubating.

Reply:
It just means you have to duplicate the weaver config yaml in both places for now

Reply:
Ah yeah would be good to DRY that up